### PR TITLE
refactor!: add template blocks for base, detail and form templates.

### PR DESCRIPTION
**Pull request by HannesWaechter**
changes:

- base.html
  - wrapped messages and content in div.container

- messages.html
  - removed div.container

- detail.html
  - use main tag instead of section
  - removed div.container inside main
  - renamed blocks
    - links -> links_container
    - heading -> heading_container

- form.html
  - use main tag instead of section
  - removed div.container from
    - main
    - form_buttons_top
    - form_buttons
  - moved mb-2 from div.container to btn-group in
    - form_buttons_top
    - form_buttons
  - removed section around from_buttons
  - renamed blocks
    - links -> links_container
    - heading -> heading_container
    - form_buttons_top -> form_buttons_top_container
    - form_errors -> form_errors_container

- create.html
  - removed h1 from block heading (is now outside)

- partials/layout.html
  - refactored col-xy -> col
  - moved row from field to detail_field_container
  - removed col from html_field_container
  - switched .text-md-right for .text-sm-right

- partials/form_layout.html
  - refactored col-xy -> col
  - moved .row from field to detail_field_container
  - moved .row .field-group to form_field_container
  - removed .col-form-label from detail_field_label
  - removed .form-group and .form-control-plaintext from detail_field
  - removed col from html_field_container
  - switched .text-right for .text-sm-right
  - moved hidden_fields outside layout

### CODE REVIEW

**Comment by HannesWaechter** [comment]
breaking changes:

- renamed blocks
    - links -> links_container
    - heading -> heading_container
    - form_buttons_top -> form_buttons_top_container
    - form_errors -> form_errors_container

File: src/beam/themes/bootstrap4/templates/beam/partials/form_layout.html [hunk 1]
```diff
@@ -1,63 +1,119 @@
...
+                        </div>
+                        {% endblock %}
+
+                    {% else %}
+
+                        {% get_form_field form field as form_field %}
+                        {% if form_field is None %}
+
+                            {% block detail_field_container %}
+                            <div class="row field-group2">
```
Comment (RaphaelKimmig) [inline src/beam/themes/bootstrap4/templates/beam/partials/form_layout.html:42]: group2?

File: src/beam/themes/bootstrap4/templates/beam/partials/form_layout.html [hunk 2]
```diff
@@ -1,63 +1,119 @@
...
+                                <div class="col-sm-4 text-sm-right beam-field-label-{{ field }} ">
+
+                                    {% block detail_field_label %}
+                                    <label class="col-form-label2">
```
Comment (RaphaelKimmig) [inline src/beam/themes/bootstrap4/templates/beam/partials/form_layout.html:46]: label2?

**Review by RaphaelKimmig — COMMENTED** [review]
- we should apply breaking block in all templates, that is 
  - Some templates still use `block heading` with an `h1` member
  - The list view still uses the old `block links`
- i don't see a `form_errors_container`
- why do we switch .text-md-right for .text-sm-right?

**Pull request merged**